cmd/promtool/main_test.go

```diff
@@ -194,7 +194,78 @@ func TestCheckTargetConfig(t *testing.T) {
 	}
 	for _, test := range cases {
 		t.Run(test.name, func(t *testing.T) {
-			_, err := checkConfig(false, "testdata/"+test.file)
+			_, err := checkConfig(false, "testdata/"+test.file, false)
+			if test.err != "" {
+				require.Equalf(t, test.err, err.Error(), "Expected error %q, got %q", test.err, err.Error())
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestCheckConfigSyntax(t *testing.T) {
+	cases := []struct {
+		name       string
+		file       string
+		syntaxOnly bool
+		err        string
+	}{
+		{
+			name:       "check with syntax only succeeds with nonexistent rule files",
+			file:       "config_with_rule_files.yml",
+			syntaxOnly: true,
+			err:        "",
+		},
+		{
+			name:       "check without syntax only fails with nonexistent rule files",
+			file:       "config_with_rule_files.yml",
+			syntaxOnly: false,
+			err:        "\"testdata/non-existent-file.yml\" does not point to an existing file",
+		},
+		{
+			name:       "check with syntax only succeeds with nonexistent service discovery files",
+			file:       "config_with_service_discovery_files.yml",
+			syntaxOnly: true,
+			err:        "",
+		},
+		// The test below doesn't fail because the file verification for ServiceDiscoveryConfigs doesn't fail the check if
+		// file isn't found; it only outputs a warning message.
```

Comment:
This makes me think a few things:
1. I wonder why checking sd files only prints a warning instead of failing like other places
2. printing a warning makes it hard to test, it might be better to have a method that returns errs/warnings to make testing easier. (I don't think that change needs to be made in this PR, just thinking out loud here)
3. I'm curious if returning warnings for failing file checks would be a better way to handle the syntax check instead of a flag. For example, if the default functionality of `checkConfig` would never  fail, but rather print any errs and the end user could decide how/if to fix.  Again just thinking out loud, I think your changes here look good, but more just having a discussion about alternatives.

Reply:
> 1. I wonder why checking sd files only prints a warning instead of failing like other places

That's a good question. I definitely think it should be consistent with other, similar file checks in the code.

> 2. printing a warning makes it hard to test, it might be better to have a method that returns errs/warnings to make testing easier. (I don't think that change needs to be made in this PR, just thinking out loud here)

Agreed. Coverage is pretty low overall for `main.go` (~30% of statements according to my last coverage run). A refactor and accompanying tests would go a long way for maintainability.

> 3. I'm curious if returning warnings for failing file checks would be a better way to handle the syntax check instead of a flag.

I'm thinking either way would be viable. Since CI is a significant use case for this tool, printing warnings would be better default behavior than just failing. We could even add a quiet flag to suppress the warnings.

I'll create an Issue for the first point, since consistency is important. I think I'll also add a point about changing the default behavior (i.e. print warnings over failing the check) to open up discussion.